src/renderer/src/components/TransactionActivityList/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { useEffect, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import { TbAlertTriangle } from 'react-icons/tb'
+import { Separator } from '@renderer/components/Separator'
+import { Skeleton } from '@renderer/components/Skeleton'
+import { AccountHelper } from '@renderer/helpers/AccountHelper'
+import { useAccountsSelector } from '@renderer/hooks/useAccountSelector'
+import { useActions } from '@renderer/hooks/useActions'
+import { useGetFullTransactions } from '@renderer/hooks/useGetFullTransactions'
+import { useInfiniteScroll } from '@renderer/hooks/useInfiniteScroll'
+import { TFullTransactionsItem } from '@shared/@types/hooks'
+import { IAccountState } from '@shared/@types/store'
+import { useVirtualizer } from '@tanstack/react-virtual'
+import { format, startOfMonth } from 'date-fns'
+import { match } from 'ts-pattern'
+
+import { TransactionActivityListItem } from './TransactionActivityListItem'
+
+type TActionsData = {
+  accounts: IAccountState[]
+}
+
+type TGroupedDataByDate = {
+  date: string
+  items: TFullTransactionsItem[]
+}
+
+type TProps = {
+  defaultAccount?: IAccountState
+}
+
+const heights = {
+  DATE: 40,
+  DATE_GAP: 16,
+  HEADER: 34,
+  EVENT: 50,
+  SEPARATOR: 1,
+  SEPARATOR_MARGIN: 8,
+  TRANSACTION_GAP: 16,
+}
+
+export const TransactionActivityList = ({ defaultAccount }: TProps) => {
+  const { t } = useTranslation('components', { keyPrefix: 'transactionActivityList' })
+  const { accounts } = useAccountsSelector()
+
+  const { actionData } = useActions<TActionsData>({
+    accounts: defaultAccount ? accounts.filter(AccountHelper.predicate(defaultAccount)) : accounts,
+  })
+
+  const dateNow = new Date()
+  const dateFrom = startOfMonth(dateNow).toISOString()
+  const dateTo = dateNow.toISOString()
+
+  const { data, isLoading, fetchNextPage } = useGetFullTransactions({ ...actionData, dateFrom, dateTo })
+  const { handleScroll, ref: scrollRef } = useInfiniteScroll<HTMLDivElement>(fetchNextPage)
+
+  const groupedDataByDates = useMemo<TGroupedDataByDate[]>(() => {
+    const groupedData: TGroupedDataByDate[] = []
+
+    data.forEach(item => {
+      const formattedDate = format(item.date, 'dd MMM yyyy')
+      const currentGroupedData = groupedData.find(({ date }) => date === formattedDate)
+
+      if (!currentGroupedData) groupedData.push({ date: formattedDate, items: [item] })
+      else currentGroupedData.items.push(item)
+    })
+
+    return groupedData
+  }, [data])
```

Comment:
Just complementing the comment in useGetFullTransaction.

We can do it within the useGetFullTransaction memo since this hook is only used here and avoid to create a new useMemo